cpp/src/plasma/malloc.cc

```diff
@@ -40,7 +44,12 @@ int fake_munmap(void*, int64_t);
 #define USE_DL_PREFIX
 #define HAVE_MORECORE 0
 #define DEFAULT_MMAP_THRESHOLD MAX_SIZE_T
-#define DEFAULT_GRANULARITY ((size_t)128U * 1024U)
+
+#ifndef ARROW_PLASMA_HUGEPAGES
+#define DEFAULT_GRANULARITY ((size_t)128U * 1024U)  // 128KB
```

Comment:
The code is refactored and this is set at runtime now. Let's not expose it to the user via a configuration option yet, it doesn't seem that there would be benefits that justify the additional complexity. I made it 1GB in the huge page case, it seems unlikely that anybody would use huge pages if they have < 1GB of memory. Happy to revisit this in the future if desired.